tests/ga/test_monitor.py

```diff
@@ -521,36 +534,82 @@ def test_send_extension_metrics_telemetry_for_empty_cgroup(self, patch_report_al
 
     @skip_if_predicate_false(are_cgroups_enabled, "Does not run when Cgroups are not enabled")
+    @skip_if_predicate_true(is_trusty_in_travis, "Does not run on Trusty in Travis")
+    @patch('azurelinuxagent.common.event.EventLogger.add_metric')
     @patch('azurelinuxagent.common.event.EventLogger.add_event')
     @attr('requires_sudo')
-    def test_send_extension_metrics_telemetry_with_actual_cgroup(self, patch_add_event, *args):
+    def test_send_extension_metrics_telemetry_with_actual_cgroup(self, patch_add_event, patch_add_metric, *arg):
         self.assertTrue(i_am_root(), "Test does not run when non-root")
 
-        num_polls = 5
-        name = "test-cgroup"
-
-        cgs = make_new_cgroup(name)
+        max_num_polls = 5
+        time_to_wait = 3
+        extn_name = "foobar-1.0.0"
 
+        cgs = make_new_cgroup(extn_name)
         self.assertEqual(len(cgs), 2)
 
-        for cgroup in cgs:
-            CGroupsTelemetry.track_cgroup(cgroup)
-
-        for i in range(num_polls):
-            CGroupsTelemetry.poll_all_tracked()
-            consume_cpu_time()  # Eat some CPU
-            consume_memory()
+        ext_handler_properties = ExtHandlerProperties()
+        ext_handler_properties.version = "1.0.0"
+        ext_handler = ExtHandler(name='foobar')
+        ext_handler.properties = ext_handler_properties
+        ext_handler_instance = ExtHandlerInstance(ext_handler=ext_handler, protocol=None)
+        ext_handler_instance.set_operation("Enable")
 
         monitor_handler = get_monitor_handler()
         monitor_handler.init_protocols()
         monitor_handler.last_cgroup_polling_telemetry = datetime.datetime.utcnow() - timedelta(hours=1)
         monitor_handler.last_cgroup_report_telemetry = datetime.datetime.utcnow() - timedelta(hours=1)
+
+        command = self.create_script("keep_cpu_busy_and_consume_memory_for_5_seconds", '''
+nohup python -c "import time
+
+for i in range(5):
+    x = [1, 2, 3, 4, 5] * (i * 1000)
+    time.sleep({0})
+    x = [1, 2, 3, 4, 5] * (i * 1000)
+    x *= 0
+    print('Test loop')" &
+'''.format(time_to_wait))
+
+        self.log_dir = os.path.join(self.tmp_dir, "log")
+
+        with patch("azurelinuxagent.ga.exthandlers.ExtHandlerInstance.get_base_dir", lambda *_: self.tmp_dir) as \
+                patch_get_base_dir:
+            with patch("azurelinuxagent.ga.exthandlers.ExtHandlerInstance.get_log_dir", lambda *_: self.log_dir) as \
+                    patch_get_log_dir:
+                ext_handler_instance.launch_command(command)
+
+        self.assertTrue(CGroupsTelemetry.is_tracked(os.path.join(
+            BASE_CGROUPS, "cpu", "walinuxagent.extensions", "foobar_1.0.0")))
+        self.assertTrue(CGroupsTelemetry.is_tracked(os.path.join(
+            BASE_CGROUPS, "memory", "walinuxagent.extensions", "foobar_1.0.0")))
+
+        for i in range(max_num_polls):
+            metrics = CGroupsTelemetry.poll_all_tracked()
+            time.sleep(0.1)
```

Comment:
Why is this sleep needed?

Reply:
Not necessary now. Removing it. Thanks for the catch.